app/templates/components/country-chooser.hbs

```diff
@@ -2,7 +2,7 @@
   <label>{{this.label}}</label>
 
   {{item-chooser
-    items=this.codes multiple=@multiple placeholder=@placeholder searchField="name"
-    selection=this._selection class=(if @dark "item-chooser--dark") size=@size
-    onSelect=this.onCountrySelection}}
+    items=this.codes multiple=this.multiple placeholder=this.placeholder searchField="name"
+    selection=this._selection class=(if this.dark "item-chooser--dark") size=this.size
+    onSelect=this.args.onCountrySelection}}
```

Comment:
Why the use of `this` instead of `@` it's style recommendation ?

Reply:
@tgallice because there is now a difference between the properties passed to the component (used via `@property` or `this.args.property`)  and the ones scoped to the component (the getters or variables) which we access with `this.myVariable`

Reply:
Have you a documentation ? This seems confusing to me ^^

Reply:
Yep, here you go ! https://guides.emberjs.com/v3.18.0/upgrading/current-edition/templates/